### PR TITLE
[FrameworkBundle][Yaml] Remove some unused code

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -2401,10 +2401,6 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
 
     private function registerRetryableHttpClient(array $options, string $name, ContainerBuilder $container)
     {
-        if (!class_exists(RetryableHttpClient::class)) {
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:2399]: `symfony/framework-bundle conflicts with `symfony/http-client < 5.4`

File: src/Symfony/Component/Yaml/Command/LintCommand.php [hunk 2]
```diff
@@ -91,10 +91,6 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->format = $input->getOption('format');
         $flags = $input->getOption('parse-tags');
 
-        if ('github' === $this->format && !class_exists(GithubActionReporter::class)) {
```
Comment (fancyweb) [inline src/Symfony/Component/Yaml/Command/LintCommand.php:94]: `symfony/yaml` conflicts with `symfony/console < 5.4`

**Review by derrabus — APPROVED** [review]
For 6.3?

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Pull request merged**